pkg/transparentproxy/iptables/builder/builder_table_nat.go

```diff
@@ -127,82 +155,137 @@ func buildMeshOutbound(
 		//     listener#[fd00::0:10]:7777 ⤸
 		//     cluster#localhost:7777 ⤸
 		//   localhost:7777
-		AddRule(
-			Source(Address(inboundPassthroughSourceAddress)),
-			OutInterface(cfg.LoopbackInterfaceName),
-			Jump(Return()),
-		).
-		AddRule(
-			Protocol(Tcp(NotDestinationPortIfBool(shouldRedirectDNS, DNSPort))),
-			OutInterface(cfg.LoopbackInterfaceName),
-			NotDestination(localhost),
-			Match(Owner(Uid(uid))),
-			Jump(ToUserDefinedChain(inboundRedirectChainName)),
-		).
-		AddRule(
-			Protocol(Tcp(NotDestinationPortIfBool(shouldRedirectDNS, DNSPort))),
-			OutInterface(cfg.LoopbackInterfaceName),
-			Match(Owner(NotUid(uid))),
-			Jump(Return()),
-		).
-		AddRule(
-			Match(Owner(Uid(uid))),
-			Jump(Return()),
+		AddRules(
+			rules.
+				NewRule(
+					Source(Address(inboundPassthroughSourceAddress)),
+					OutInterface(cfg.LoopbackInterfaceName),
+					Jump(Return()),
+				).
+				WithCommentf("prevent traffic loops by ensuring traffic from the sidecar proxy (using %s) to loopback interface is not redirected again", inboundPassthroughSourceAddress),
+			// Redirect outbound TCP traffic (except DNS port 53) from the
```

Comment:
Is it necessary to have the comment here in the code and in the rules?

Reply:
Addressed